### DISCUSSION TITLE
Schedule a playlist for once a month

### DISCUSSION THREAD

**Discussion by ChandigarIT** [AzuraCast/AzuraCast#7618] [Ideas]
At the moment, we can schedule a playlist by days of the week. We have DJs who produce a show once a month, and it would be useful to be able to schedule for something like "1st Friday, 4th Friday, or LAST Friday of the month". Is this something that could be implemented?

Many thanks.

**Comment by emtoonst** [comment]
You could play with the Liquidsoap configuration:

Add the following logic to your Liquidsoap configuration file to check for the specific Friday:

```
def is_specific_friday(year, month, day, nth_friday)
  weekday = date.weekday(year, month, day)  # 0 = Monday, ..., 6 = Sunday
  if weekday != 4 then
    false  # Not a Friday
  else
    week_of_month = int((day - 1) / 7) + 1
    total_days = date.days_in_month(year, month)
    last_week_of_month = int((total_days - 1) / 7) + 1
    if nth_friday == week_of_month or (nth_friday == -1 and week_of_month == last_week_of_month) then
      true
    else
      false
    end
  end
end
```

Create the predicates:

```
predicate.first_friday = predicate.custom(is_specific_friday(date.year(), date.month(), date.day(), 1))
predicate.fourth_friday = predicate.custom(is_specific_friday(date.year(), date.month(), date.day(), 4))
predicate.last_friday = predicate.custom(is_specific_friday(date.year(), date.month(), date.day(), -1))
```

And use those predicates to schedule the playlists:

```
radio = fallback(track_sensitive=false, [
    switch([
        (predicate.first_friday, playlist_first_friday_show),
        (predicate.fourth_friday, playlist_fourth_friday_show),
        (predicate.last_friday, playlist_last_friday_show)
    ]),
    radio
])
```

**Reply by ChandigarIT** [reply]
At the "Create the predicates" I am getting the following message.

![image](https://github.com/user-attachments/assets/57b26dfe-9e84-4b6e-ac71-c0bce1981ecf)

predicate.first_friday **{HERE}** = predicate.custom(is_specific_friday(date.year(), date.month(), date.day(), 1))

**Reply by gAlleb** [reply]
I've been avoiding from comments to this topic)) Maybe my Liquidsoap knowledge is not enough.

Of course it would give you all kind of errors because it looks like a AI hallucinated code - not a real one)

**Comment by emtoonst** [comment]
It is AI hallucinated code, but the concept should work, but you have to use AzuraCast's supported Liquidsoap time functions instead of the predicates. I have a similar implementation and it works perfectly.

**Comment by ChandigarIT** [comment]
Still no joy. 

Are you able to share what you have implemented? Thanks.


(Also 1 = Monday) https://liquidsoap.readthedocs.io/en/latest/content/language.html

**Comment by eternityready2** [comment]
> At the moment, we can schedule a playlist by days of the week. We have DJs who produce a show once a month, and it would be useful to be able to schedule for something like "1st Friday, 4th Friday, or LAST Friday of the month". Is this something that could be implemented?
> 
> Many thanks.

@ChandigarIT @emtoonst @gAlleb 

I have version of azura cast where I built extended scheduling options where once a month is included. It also has AI DJS and other cool features.  Its available from my git hub @ https://github.com/eternityready2/AzuraCast

**Comment by ChandigarIT** [comment]
Many thanks. I have passed this on to the web hoster.

Malcolm

**Reply by eternityready2** [reply]
@ChandigarIT What do you mean you have passed this onto web hoster? which web host is it?

**Comment by ChandigarIT** [comment]
I used to program the shows for TopCat, hosted by someone in spain. TopCat
is no longer on air, but there is another station that is on the same
server, and programmed by someone else, so he will find this very useful
for the other stations that he helps run.

Thank you
